src/parse.y

```diff
@@ -984,13 +983,8 @@ expr:
 					}
 				else
 					{
-					if ( out_of_scope_locals.count(id.get()) > 0 &&
-					     warned_about_locals.count(id.get()) == 0 )
-						{
-						// Remove in v5.1
-						reporter->Warning("use of out-of-scope local %s deprecated; move declaration to outer scope", id->Name());
-						warned_about_locals.insert(id.get());
-						}
+					if ( out_of_scope_locals.count(id.get()) > 0 )
+						id->Error("use of out-of-scope local; move declaration to outer scope");
```

Comment:
Just a comment: I think this is fine for now.  One day we might want to switch to truly removing the local from the scope, once its scripting scope terminates (which would allow redefinition of the local in a subsequent scope, as is supported by most common languages).